tlslite/handshakesettings.py

```diff
@@ -284,11 +285,14 @@ def _sanityCheckProtocolVersions(other):
         """Check if set protocol version are sane"""
         if other.minVersion > other.maxVersion:
             raise ValueError("Versions set incorrectly")
-        if other.minVersion not in ((3, 0), (3, 1), (3, 2), (3, 3)):
+        if other.minVersion not in KNOWN_VERSIONS:
             raise ValueError("minVersion set incorrectly")
-        if other.maxVersion not in ((3, 0), (3, 1), (3, 2), (3, 3), (3, 4)):
+        if other.maxVersion not in KNOWN_VERSIONS:
             raise ValueError("maxVersion set incorrectly")
 
+        if other.maxVersion < (3, 4):
+            other.versions = [i for i in other.versions if i < (3, 4)]
+
```

Comment:
Couldn't this be generalized for any versions? e.g.,
```python
other.versions = [i for i in other.versions if i < other.maxVersions]
```

Reply:
it could, but addition of any new protocol version requires a lot of changes and this is obsoleted later by addition of server side support

Reply:
correction, it couldn't because `other.versions` can include draft version `(127, 26)`; that generalization would work only once the protocol is finalized